codegen/java-gen/src/test/scala/com/lightbend/akkasls/codegen/java/ActionTestKitGeneratorSuite.scala

```diff
@@ -0,0 +1,165 @@
+          |import com.akkaserverless.javasdk.testkit.impl.ActionResultImpl;
+          |import com.akkaserverless.javasdk.testkit.impl.StubActionContext;
+          |import com.akkaserverless.javasdk.testkit.impl.StubActionCreationContext;
+          |import com.example.actions.CounterJournalToTopicAction;
+          |import com.example.actions.CounterTopicApi;
+          |import com.example.domain.CounterDomain;
+          |import com.google.protobuf.Any;
+          |import com.google.protobuf.Empty;
+          |import java.util.ArrayList;
+          |import java.util.List;
+          |import java.util.Optional;
+          |import java.util.function.Function;
+          |
+          |public final class CounterJournalToTopicActionTestKit {
+          |
+          |  private Function<ActionCreationContext, CounterJournalToTopicAction> actionFactory;
+          |
+          |  private CounterJournalToTopicAction createAction() {
+          |    CounterJournalToTopicAction action = actionFactory.apply(new StubActionCreationContext());
+          |    action._internalSetActionContext(Optional.of(new StubActionContext()));
+          |    return action;
+          |  };
+          |
+          |  public static CounterJournalToTopicActionTestKit of(Function<ActionCreationContext, CounterJournalToTopicAction> actionFactory) {
+          |    return new CounterJournalToTopicActionTestKit(actionFactory);
+          |  }
+          |
+          |  private CounterJournalToTopicActionTestKit(Function<ActionCreationContext, CounterJournalToTopicAction> actionFactory) {
+          |    this.actionFactory = actionFactory;
+          |  }
+          |
+          |  private <E> ActionResult<E> interpretEffects(Action.Effect<E> effect) {
+          |    return new ActionResultImpl(effect);
+          |  }
+          |
+          |  public ActionResult<CounterTopicApi.Increased> increase(CounterDomain.ValueIncreased event) {
+          |    Action.Effect<CounterTopicApi.Increased> effect = createAction().increase(event);
+          |    return interpretEffects(effect);
+          |  }
+          |
+          |  public ActionResult<CounterTopicApi.Decreased> decrease(CounterDomain.ValueDecreased event) {
+          |    Action.Effect<CounterTopicApi.Decreased> effect = createAction().decrease(event);
+          |    return interpretEffects(effect);
+          |  }
+          |
+          |  public ActionResult<Empty> ignore(CounterDomain.Any event) {
+          |    Action.Effect<Empty> effect = createAction().ignore(event);
+          |    return interpretEffects(effect);
+          |  }
+          |
+          |}""".stripMargin
+
+    assertNoDiff(sourceCode, expected)
+  }
+
+  test("it can generate an specific Test stub for the entity") {}
+
+  /**
+   * This ModelBuilder.EventSourcedEntity is equivalent to the entity in
+   * test/resources/testkit/shoppingcart_domain.proto
+   */
+  def generateModel(): Try[ModelBuilder.Model] = {
+
```

Comment:
This is a new approach to create the initial BuilderModel.Model by reading the user-function.desc file from one of the samples. Then we can inspect its content by printing it with munit.Assertions.munitPrint(model).

Reply:
Ok, was it added because that is easier than to build up a realistic model manually like the TestData?

Reply:
yep. I find it easier and less prone to error.